spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkCatalog.java

```diff
@@ -43,4 +49,8 @@ public Table loadTable(Identifier ident) throws NoSuchTableException {
 
     return new SparkTable(table, false);
   }
+
+  public static void setDummyIcebergTbl(Table dummyTbl) {
```

Comment:
Probably not needed, but you could do the similar map of "Identifier -> Table" and just allow our tests to just insert new tables into that map and then request them on demand. Then have a method clear it out after each test. Just thinking about this for tests which run in multiple parameterizations (which will have the same static dummyIcebergTbl)

I know i'm biased since this is how I did it my other example, I like your placement in the Catalog here much more than mine though.

Reply:
+1, Agree with you, present impl allows me to set only 1 tbl per UT, having a map and doing a look-up can allow me creating more than 1 tbl at given time :) !!!

Made the changes